sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample08_CustomEventProcessor.md

```diff
@@ -0,0 +1,439 @@
+
+## Building an event processor with EventProcessor&lt;TPartition&gt;
+
+The [EventProcessor&lt;TPartition&gt;](https://docs.microsoft.com/dotnet/api/azure.messaging.eventhubs.primitives.eventprocessor-1) serves as a base for creating a custom event processor.  It fills a role similar to the `EventProcessorClient`, allowing you to read from all partitions of an Event Hub with cooperative load balancing and a high degree of resiliency.  `EventProcessor<TPartition>` also offers native batch processing, the ability to customize data storage, and a greater level of control over communication with the Event Hubs service.  `EventProcessor<TPartition>` is an abstract class, which has five methods you'll need to provide implementations for.  If you've used the `EventProcessorClient`, the first two will be familiar to you:
+
+- `OnProcessingEventBatchAsync`: The actual "business logic" of your processor.  This is similar to the `ProcessEvent` event exposed by `EventProcessorClient`.
+- `OnProcessingErrorAsync`: A handler that can be used to observe exceptions from inside the machinery of the EventProcessor itself.  This is similar to the `ProcessError` event exposed by `EventProcessorClient`.
+
+The remaining three deal with checkpointing and load balancing:
+
+- `ListOwnershipAsync`: Used by load balancing to see which partitions have been assigned to which processors.
+- `ClaimOwnershipAsync`: Used by load balancing to grab ownership of a partition (which may or may not be assigned to another processor).
+- `ListCheckpointsAsync`: Exists only for backwards compatibility and is called only when the `GetCheckpointAsync` method is not overridden.  For more efficient operation, it is recommended to implement `GetCheckpointAsync` and throw from `ListCheckpointsAsync`
+
+There are also three methods that you are not required to implement, but may find useful to do so:
+
+- `GetCheckpontAsync`: Used to query storage for a specific checkpoint.  This method is meant to supersede `ListCheckpointsAsync` and we recommend implementing it for efficiency.
+
+- `OnInitializingPartitionAsync`: A handler that can be used to observe when the event processor takes ownership of a partition and is preparing to start processing its events.
+
+- `OnPartitionProcessingStoppedAsync`: A handler that can be used to observe when the event processor loses ownership of a partition and is no longer processing its events.
+
+The `On*` methods can only be implemented when we know what we want our event processor to do, but the remaining can be implemented independent of the business logic for processing events.
```

Comment:
This sentence is unclear to me. Is there an example that would be relevant for a scenario when we know what we want our event processor to do?